### DISCUSSION TITLE
`openapi-typescript` doesn't set the type as `number` if column is integer. Is this expected?

### DISCUSSION THREAD

**Discussion by baptisteArno** [supabase/supabase#3143] [Questions]
I have a table with an `int2` column and when generating the types with `openapi-typescript` it sets the field as a string. It should be a number, right? Is there anything I can do?

**Comment by steve-chavez** [comment]
As a workaround, you can modify the openapi output yourself with a js script - see it on https://github.com/supabase/supabase/discussions/984#discussioncomment-1165328

**Comment by steve-chavez** [comment]
You could also patch PostgREST, from where the OpenAPI output is generated(currently), here's the part where the mapping from pg to openapi types is done:

https://github.com/PostgREST/postgrest/blob/main/src/PostgREST/OpenAPI.hs#L57-L69
